profile/src/main/java/rdx/works/profile/domain/GetProfileUseCase.kt

```diff
@@ -28,20 +30,15 @@ class GetProfileUseCase @Inject constructor(
     }
 
     /**
-     * Checks the validity of the profile. A profile might have been temporarily generated, but might contain no accounts.
-     * This is considered as a profile that is not properly initialized, as a correct profile should have at least one account,
-     * meaning at least one network.
+     * Checks the validity of the profile. A profile might have been temporarily generated, but might contain no networks.
+     * This is considered as a profile that is not properly initialized, as a correct profile should have at least one network.
      *
      * This method is crucial for backing up the profile, since we don't want to sync profile to Drive when profile is
      * not initialized yet. For example, when user adds a ledger to create an account in onboarding flow.
-     *
      */
-    suspend fun isInitialized(): Boolean = profileRepository.profileState
-        .firstOrNull()?.let { state ->
-            isInitialized(state)
-        } == true
-
-    private fun isInitialized(state: ProfileState): Boolean {
-        return state is ProfileState.Restored && state.hasNetworks()
-    }
+    suspend fun finishedOnboardingProfile(): Profile? = profileRepository.profileState
```

Comment:
why not `initializedProfile()` or `getInitializedProfile()`? I find it kinda misleading 😄

Reply:
the following is just thoughts - actions not really needed.
I know we already discussed this...

Now, we actually have two functions to get the `Profile` and this is a bit confusing(?). 

I do understand the difference, and I am thinking what if we decide to keep only one (the nullable). But that would cause a lot of nullable checks...

And then what if we keep the `invoke()` and add this piece of code (finishedOnboardingProfile) inside the invoke and throw an exception. Then in the places we must have the initialized profile we catch the error.